arch/arm/src/nrf52/nrf52_ppi.c

```diff
@@ -0,0 +1,134 @@
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include "arm_arch.h"
+#include "chip.h"
+#include "nrf52_ppi.h"
+#include "hardware/nrf52_ppi.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nrf52_ppi_channel_enable
+ ****************************************************************************/
+
+void nrf52_ppi_channel_enable(uint8_t ch, bool enable)
+{
+  DEBUGASSERT(ch < NRF52_PPI_NUM_CHANNELS);
+
+  if (enable)
+    {
+      putreg32(PPI_CHEN_CH(ch), NRF52_PPI_CHENSET);
+    }
+  else
+    {
+      putreg32(PPI_CHEN_CH(ch), NRF52_PPI_CHENCLR);
+    }
+}
+
+/****************************************************************************
+ * Name: nrf52_ppi_set_event_ep
+ ****************************************************************************/
+
+void nrf52_ppi_set_event_ep(uint8_t ch, volatile uint32_t *event_reg)
```

Comment:
Why not just 'uint32_t event_reg' ?

Reply:
The same applies to the rest of the functions that take as an argument register address.

Reply:
As these are register addresses, I understand they should be declared as volatile to avoid any caching introduced by the compiler. But I'm not sure if it is actually required in this case.

Reply:
This is done in the putreg32 macro itself, so the arguments can be just uint32_t.
From arch/arm/src/common/arm_arch.h:
`# define putreg32(v,a)        (*(volatile uint32_t *)(a) = (v))`

Reply:
You mean `uint32_t*`? Or `uint32_t` directly?

Reply:
I mean `uint32_t` directly. I suppose these functions will be called with the address definitions from nrf52/hardware which are just memory offsets with no type defined, so there should be no problems.

Reply:
done